source/common/grpc/async_client_impl.cc

```diff
@@ -141,11 +141,17 @@ void AsyncStreamImpl::onHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_s
 
 void AsyncStreamImpl::onData(Buffer::Instance& data, bool end_stream) {
   decoded_frames_.clear();
-  if (!decoder_.decode(data, decoded_frames_)) {
+  auto status = decoder_.decode(data, decoded_frames_);
+  if (status.code() == absl::StatusCode::kInternal) {
     streamError(Status::WellKnownGrpcStatus::Internal);
     return;
   }
 
+  if (status.code() == absl::StatusCode::kResourceExhausted) {
+    streamError(Status::WellKnownGrpcStatus::ResourceExhausted);
```

Comment:
What about other errors in future? Should all error just be mapped into stream errors and early return?

Reply:
Currently only two Internal and ResourceExhausted are returned from decode().

But forward looking, it is a good point. Done.